src/core/abr/representation_chooser.ts

```diff
@@ -449,8 +501,39 @@ export default class RepresentationChooser {
             nextBitrate /= clock.speed;
           }
 
+          if ((this.currentDroppedFrames / this.currentTotalFrames) > 0.33) {
+            if (clock.bitrate) {
+              const decodableBitrate =
+                getDecodableBitrate(representations, clock.bitrate);
+              if (decodableBitrate < this._decodableBitrate) {
+                this._decodableBitrate = decodableBitrate;
+                if (this._timeOutId !== null) {
+                  clearTimeout(this._timeOutId);
+                }
+                this._timeOutId = window.setTimeout(() => {
```

Comment:
something still irks me about the setTimeout part. I feel like this could be better implemented with RxJS operators, and make use of the containing observable and of ``this._dispose`` in place of adding 2 private variable to the class dedicated to this call.